### PR TITLE
Remove some dead code from transaction enrichment

### PR DESCRIPTION
**Pull request by hurryabit**
When trying to understand how disclosure/divulgence information is
computed for the scneario service, I stumbled on some dead code:

* The `EnrichState.divulgeContracts` function is never used. Let's
  remove it.

* The `EnrichState.localDivulgendes` field is initialized with an empty
  map and never touched again. Thus, it's always going to stay an empty
  map. This means that the
  `EnrichedTransaction.localImplicitDisclosure` is also always going to
  be an empty map. Both fields can hence be removed.

I'm not sure if the `BlindingInfo.localDivulgence` field can be removed
as well. In my understanding, this data structure could be loaded from
transactions serialized with the field being non-empty in the past.
Thus, I've refrained from removing the field and set it to an empty map
when constructing a `BlindingInfo` from an `EnrichedTransaction`.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/6507)

### CODE REVIEW

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala [hunk 1]
```diff
@@ -50,9 +50,12 @@ object Blinding {
     if (enrichedTx.failedAuthorizations.isEmpty) {
       Right(
         BlindingInfo(
-          enrichedTx.explicitDisclosure,
-          enrichedTx.localImplicitDisclosure,
-          enrichedTx.globalImplicitDisclosure))
+          disclosure = enrichedTx.explicitDisclosure,
+          // NOTE(MH): Local divulgences are a thing of the past and are not
+          // produced by the current implementation anymore.
+          localDivulgence = Map.empty,
```
Comment (hurryabit) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:56]: I don't know if we could be more aggressively clean up here and would need some advice on this. If we can clean up more, I'd leave for a separate PR nevertheless.

Comment (remyhaemmerle-da) [reply daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:56]: I thinks @gerolf-da (or maybe @stefanobaghino-da) would be better at giving advice on this.

Comment (stefanobaghino-da) [reply daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:56]: I see only four lines that use `localDivulgence`. You can probably safely remove it and keep this PR very small. In any case, I would prefer if you could at least put this so that it's an invariant at construction, rather than a comment here where it's called.

Comment (gerolf-da) [reply daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:56]: Yes, with the removal of `RelativeContractId`, `localDivulgence` lost it's justification to be a thing.

Comment (stefanobaghino-da) [reply daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:56]: [This](https://github.com/digital-asset/daml/pull/6509) looks like it would do it, testing if it doesn't breaking anything.

Comment (hurryabit) [reply daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:56]: @stefanobaghino-da I'm fine with merging your suggestion in once we've discussed my comment on it.

Comment (stefanobaghino-da) [reply daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:56]: Ran all tests under `//daml-lf` and one conformance test, all good, unfortunately I cannot run the full thing on CI unless I'm trying to merge on `master`.

Comment (stefanobaghino-da) [reply daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:56]: > Yes, with the removal of `RelativeContractId`, `localDivulgence` lost it's justification to be a thing.

@remyhaemmerle-da is crying of joy.

Comment (hurryabit) [reply daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala:56]: Since it's still unclear if we can apply @stefanobaghino-da's suggestion without breaking backward compatibility, we merge this as is. Because of backward compatibility concerns, we can also not address
> In any case, I would prefer if you could at least put this so that it's an invariant at construction, rather than a comment here where it's called.
